lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -579,33 +579,27 @@ def write(chunk)
             # Save the following fields if available, then clear them out to
             # allow for determining whether we should parse the log or message
             # field.
-            timestamp = record.delete('time')
-            severity = record.delete('severity')
-            trace = record.delete(@trace_key)
-            span_id = record.delete(@span_id_key)
-            insert_id = record.delete(@insert_id_key)
+            preserved_keys = ['time', 'severity', @trace_key, @span_id_key,
```

Comment:
[Optional] Might be more readable as:
```ruby
            preserved_keys = [
              'time', 'severity', @trace_key, @span_id_key, @insert_id_key
            ]
```
Don't know if Rubocop will like that closing brace on its own line, though, so you might have to join it onto the previous one.